sectionate/transports.py

```diff
@@ -0,0 +1,505 @@
+               if d in c}.items()}
+
+    for p in range(len(uvindices['var'])):
+        var, i, j = uvindices['var'][p], uvindices['i'][p], uvindices['j'][p]
+        if var == 'U':
+            if (f"geolon_u" in unames) and (f"geolat_u" in unames):
+                lon = ds[unames[f"geolon_u"]].isel({
+                    coords["X"]["q"]:i,
+                    coords["Y"]["h"]:j
+                }).values
+                lat = ds[unames[f"geolat_u"]].isel({
+                    coords["X"]["q"]:i,
+                    coords["Y"]["h"]:j
+                }).values
+            elif (f"geolon_q" in qnames) and (f"geolat_h" in hnames):
+                lon = ds[qnames[f"geolon_q"]].isel({
+                    coords["X"]["q"]:i,
+                    coords["Y"]["q"]:j
+                }).values
+                lat = ds[hnames[f"geolat_h"]].isel({
+                    coords["X"]["h"]:wrap_idx(i, grid, "X"),
+                    coords["Y"]["h"]:wrap_idx(j, grid, "Y")
+                }).values
+            else:
+                raise ValueError("Cannot locate grid coordinates necessary to\
+                identify U-velociy faces.")
+        elif var == 'V':
+            if (f"geolon_v" in vnames) and (f"geolat_v" in vnames):
+                lon = ds[vnames[f"geolon_v"]].isel({
+                    coords["X"]["h"]:wrap_idx(i, grid, "X"),
+                    coords["Y"]["q"]:j
+                }).values
+                lat = ds[vnames[f"geolat_v"]].isel({
+                    coords["X"]["h"]:wrap_idx(i, grid, "X"),
+                    coords["Y"]["q"]:j
+                }).values
+            elif (f"geolon_h" in hnames) and (f"geolat_q" in qnames):
+                lon = ds[hnames[f"geolon_h"]].isel({
+                    coords["X"]["h"]:wrap_idx(i, grid, "X"),
+                    coords["Y"]["h"]:wrap_idx(j, grid, "Y")
+                }).values
+                lat = ds[qnames[f"geolat_q"]].isel({
+                    coords["X"]["q"]:i,
+                    coords["Y"]["q"]:j
+                }).values
+            else:
+                raise ValueError("Cannot locate grid coordinates necessary to\
+                identify V-velociy faces.")
+        lons[p] = lon
+        lats[p] = lat
+    return lons, lats
+    
+def uvcoords_from_qindices(grid, isec, jsec):
+    """
+    Directly finds coordinates of velocity points from vorticity point indices, wrapping other functions.
+
+    PARAMETERS:
+    -----------
+    grid: xgcm.Grid
+        Grid object describing ocean model grid and containing data variables
+    isec: int
+        vorticity point indices along 'X' dimension 
+    jsec: int
+        vorticity point indices along 'Y' dimension
+
+    RETURNS:
+    --------
+    lons : np.ndarray(float)
+    lats : np.ndarray(float)
+    """
+    return uvcoords_from_uvindices(
+        grid,
+        uvindices_from_qindices(grid, isec, jsec),
+    )
+
+def convergent_transport(
```

Comment:
Can you describe this issue in more detail? Was this function call in one of the two OSNAP notebooks slower than you expected? I also am not sure why it would be slow because the index/mask arrays are 1D and should be really fast; the only substantial operations should be indexing the velocities, applying the masks, and adding the masks U/V:
https://github.com/raphaeldussin/sectionate/blob/550aa7c674329decb676d2830852bc197bafae5f/sectionate/transports.py#L306-L316